File: 137-SingleNumberII/137-SingleNumberII.cpp
// Last updated: 4/19/2025, 5:27:51 PM
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        vector<int> bits(32, 0);
        
        // Count bits at each position
        for (int num : nums) {
            for (int i = 0; i < 32; i++) {
                bits[i] += (num >> i) & 1;
                if (bits[i] == 3) bits[i] = 0; // Replace modulo with conditional
            }
        }
        
        int result = 0;
        for (int i = 0; i < 32; i++) {
            result |= (bits[i] << i);
        }
        
        return result;
    }
};

File: 137-SingleNumberIi/137-SingleNumberIi.py
# Last updated: 7/2/2026, 11:16:54 PM
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        bits = [0] * 32
        
        for num in nums:
            n = num & 0xFFFFFFFF if num < 0 else num
            
            for i in range(32):
                if (n >> i) & 1:
                    bits[i] += 1
                    bits[i] %= 3
        
        result = 0
        for i in range(32):
            if bits[i]:
                result |= (1 << i)
        
        if bits[31]:
            result = result - (1 << 32)
            
        return result